### DISCUSSION TITLE
Is there a way to list pages?

### DISCUSSION THREAD

**Discussion by mord0d** [xenocrat/chyrp-lite#238] [Q&A] [answered]
I mean with routes, like `tags/` for tags and `archive/` for posts. With pagination, if needed.

Or, is it possible to achieve it without messing with PHP?

**Comment by xenocrat** [comment, marked as answer]
Hello there,

There's a way to do this without writing any PHP. I'm going to do this with Virgula theme, because it's very simple and so easier to see how the solution is working - but you could do the same thing with any theme. The steps are:

1. Create a new page with the slug "index-of-pages" and a title of your choice, then save it. Page content isn't important.
2. Copy the attached Twig template, [page_index-of-pages.zip](https://github.com/xenocrat/chyrp-lite/files/13975319/page_index-of-pages.zip), to _themes/virgula/pages/page_index-of-pages.twig_.
3. Visit your new page. It will show a hierarchy of all your listed pages (it won't show pages with a status that prevents listing).
4. Optionally, set the status of your "index-of-pages" page to "Public" (but not visible in pages list). This will prevent your special index page from showing up in the hierarchy that it displays.

This works because of two features:
1. Any twig template can call the method `theme.pages_list()` to receive a list of all pages.
2. Pages support customised Twig templates with the format _page\_\<slug\>.twig_. We've hijacked this feature to create a twig that does what we want, with no PHP required.

Hope that helps!

This gives me the idea that being able to display custom Twig templates would be a useful feature. I might look into implementing this so that, in future, you won't need to create the dummy page to make it work.

**Reply by mord0d** [reply]
> Create a new page

> `pages/page_<slug>.twig`

That's it! Thank you! ☺
